drivers/peci/peci_ite_it8xxx2.c

```diff
@@ -0,0 +1,370 @@
+	 * were definitely enabled before the error occurred.
+	 * Here is the recovery mechanism for recovering the PECI
+	 * bus when the errors occur.
+	 */
+	peci_it8xxx2_set_dvie(dev, true);
+	peci_it8xxx2_init_vtts(dev, HOVTTS0P95V);
+	peci_it8xxx2_configure(dev, PECI_IT8XXX2_BITRATE_1MHZ);
+	peci_it8xxx2_enable(dev);
+	printk("[PECI] Reinitialization Finished.\r\n");
+}
+
+static int peci_it8xxx2_transfer(const struct device *dev, struct peci_msg *msg)
+{
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	struct peci_buf *peci_rx_buf = &msg->rx_buffer;
+	struct peci_buf *peci_tx_buf = &msg->tx_buffer;
+
+	int cnt, ret_code;
+
+	ret_code = 0;
+
+	if (!(peci_regs->HOCTLR & PECIHEN)) {
+		printk("[PECI] Please call the peci_enable() first.\r\n");
+		return -ECONNREFUSED;
+	}
+
+	if (peci_it8xxx2_check_host_busy(dev) != 0) {
+		return -EBUSY;
+	}
+
+	peci_regs->HOTRADDR = msg->addr;
+	peci_regs->HOWRLR = peci_tx_buf->len;
+	peci_regs->HORDLR = peci_rx_buf->len;
+	peci_regs->HOCMDR = msg->cmd_code;
+
+	if (msg->cmd_code != PECI_CMD_PING) {
+		for (cnt = 0; cnt < (peci_tx_buf->len - 1); cnt++) {
+			peci_regs->HOWRDR = peci_tx_buf->buf[cnt];
+		}
+	}
+
+	/* Host Available */
+	irq_enable(config->irq_no);
+	peci_regs->HOCTLR |= START;
+	ret_code = peci_it8xxx2_check_host_finish(dev);
+
+	if (!ret_code) {
+		/* Host Transactions Finished, Fetch Data from the regs */
+		if (peci_rx_buf->len) {
+			for (cnt = 0; cnt < (peci_rx_buf->len); cnt++) {
+				peci_rx_buf->buf[cnt] = peci_regs->HORDDR;
+			}
+		}
+		peci_it8xxx2_rst_status(dev);
+
+	} else {
+		/* Host Transactions Failure */
+		peci_it8xxx2_rst_module(dev);
+	}
+
+	return (ret_code);
+}
+
+static void peci_it8xxx2_isr(const struct device *dev)
+{
+	struct peci_it8xxx2_data *data = dev->data;
+	const struct peci_it8xxx2_config *config = dev->config;
+
+	irq_disable(config->irq_no);
+	k_sem_give(&data->device_sync_sem);
+}
+
+static const struct peci_driver_api peci_it8xxx2_driver_api = {
+	.config = peci_it8xxx2_configure,
+	.enable = peci_it8xxx2_enable,
+	.disable = peci_it8xxx2_disable,
+	.transfer = peci_it8xxx2_transfer,
+};
+
+static int peci_it8xxx2_init(const struct device *dev)
+{
+	struct peci_it8xxx2_data *data = dev->data;
+	const struct peci_it8xxx2_config *config = dev->config;
+
+	/* Initialize Semaphore */
+	k_sem_init(&data->device_sync_sem, 0, 1);
+
+	/* Configure the GPF6 to Alternative Function 3: PECI */
+	pinmux_pin_set(config->alts_list[PECI0].pinctrls,
+					config->alts_list[PECI0].pin,
+					config->alts_list[PECI0].alt_fun);
+
+	peci_it8xxx2_set_dvie(dev, true);
+	peci_it8xxx2_init_vtts(dev, HOVTTS0P95V);
+	peci_it8xxx2_configure(dev, PECI_IT8XXX2_BITRATE_1MHZ);
+
+	/* Interrupt Assignment */
+	IRQ_CONNECT(DT_INST_IRQN(0),
+	0,
+	peci_it8xxx2_isr,
+	DEVICE_DT_INST_GET(0),
+	0);
+
+	return 0;
+}
+
+DEVICE_DT_INST_DEFINE(0,
```

Comment:
I'm guessing this is a single-instance driver because there is only one such peripheral on currently supported SoCs?

If so, can you please add a `BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1, ...)` line here too, and a comment explaining what's going on?

If not, please make this a multi-instance driver; we are well past the time in the project where we should be adding single-instance drivers when there are potentially multiple hardware instances.

Reply:
Dear Marti,

Thanks for reminding me of this. Hmm, I'd like to change it but when I read the specification of PECI3.0 and the Intel platform datasheets, it seems that there's always 1-pin used in the PECI interface. Does it mean that there's probability that the PECI be extended? 

Thanks for the patience.

Best regards,
BJ

Reply:
Dear Marti,

Thanks for your guidance, and I'd like to consult you with if the following statement is appropriate to describe the current situation of this driver module.

`
/*
 * This driver is single-instance. If the devicetree contains multiple
 * instances, this will fail and the driver needs to be revisited.
 */
BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) <= 1,
	     "Unsupported PECI Instance");
`

Thanks again for your helps.

Reply:
Hi Marti,

Thanks for sharing your opinions.
I've added the following statements in the latest code:

```
/*
 * This driver is single-instance. If the devicetree contains multiple
 * instances, this will fail and the driver needs to be revisited.
 */
BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) <= 1,
	     "Unsupported PECI Instance");
```

Meanwhile, so far the latest image is also tested with the `samples/drivers/peci`.
Thanks for guidance.
BJ